sway/input/keyboard.c

```diff
@@ -282,6 +284,35 @@ static bool keyboard_execute_compositor_binding(struct sway_keyboard *keyboard,
 	return false;
 }
 
+static bool keyboard_execute_pointer_keysyms(struct sway_keyboard *keyboard,
+		const xkb_keysym_t *pressed_keysyms, size_t keysyms_len,
+		enum wl_keyboard_key_state state) {
+	struct sway_cursor *cursor = keyboard->seat_device->sway_seat->cursor;
+
+	for (size_t i = 0; i < keysyms_len; ++i) {
+		xkb_keysym_t keysym = pressed_keysyms[i];
+
+		uint32_t button = wlr_keyboard_keysym_to_pointer_button(keysym);
+		if (button != 0) {
+			dispatch_cursor_button(cursor, NULL, 0, button,
+				(enum wl_pointer_button_state)state);
+			wlr_seat_pointer_notify_frame(cursor->seat->wlr_seat);
+			return true;
+		}
+
+		int dx, dy;
+		wlr_keyboard_keysym_to_pointer_motion(keysym, &dx, &dy);
+		if (state == WL_KEYBOARD_KEY_STATE_PRESSED && (dx != 0 || dy != 0)) {
+			wlr_cursor_move(cursor->cursor, NULL, dx * 10, dy * 10);
+			cursor_rebase(cursor);
+			wlr_seat_pointer_notify_frame(cursor->seat->wlr_seat);
```

Comment:
Maybe we should just call `pointer_motion(cursor, time, &keyboard->wlr.base, dx, dy, dx, dy)`?

Reply:
Indeed, that sounds better!